src/main/java/org/pac4j/springframework/security/authentication/ClientAuthenticationProvider.java

```diff
@@ -75,7 +77,12 @@ public Authentication authenticate(final Authentication authentication) throws A
         final Client client = this.clients.findClient(clientName);
         // get the user profile
         final WebContext context = token.getContext();
-        final UserProfile userProfile = client.getUserProfile(credentials, context);
+        UserProfile userProfile = null;
+        try {
+            userProfile = client.getUserProfile(credentials, context);
+        } catch (RequiresHttpAction requiresHttpAction) {
+            throw new AuthenticationCredentialsException(requiresHttpAction.getMessage(), requiresHttpAction);
```

Comment:
In fact, no. We need to treat `RequiresHttpAction` as real HTTP actions. Maybe `return;` should be enough. For sure, it should not generate an exception. Maybe add a `TODO` for later update.